### PR TITLE
SPORTSHUB-204 adding email js to suggestions page

### PR DESCRIPTION
**Pull request by ReggieStar11**
## Why was your change needed?

Please detail here why you did the change you did.

## What was Changed?

Please detail here what were the main changes you made.

## Any extra notes for reviewers to know?

Write any extra notes you may have for the PR.
This may include specific code paths that need extra scrutiny, or how to review a specific portion of code.

### CODE REVIEW

File: frontend/observability/logger.ts [hunk 1]
```diff
@@ -1,78 +1,69 @@
...
 export class Logger {
   loggingContext: { [key: string]: string };
 
   constructor(loggerName: string, loggingContext?: { [key: string]: string }) {
-    if (loggingContext !== undefined) {
-      this.loggingContext = loggingContext;
-    } else {
-      this.loggingContext = {};
-    }
+    this.loggingContext = loggingContext || {};
```
Comment (brian2w) [inline frontend/observability/logger.ts:8]: I presume these changes are from another branch?

File: frontend/app/(footer)/suggestions/page.tsx [hunk 2]
```diff
@@ -1,66 +1,117 @@
...
+
+    try {
+      await sendEmail(templateParams);
+      alert("Thank you for your feedback!");
+      setFirstName("");
+      setLastName("");
+      setEmail("");
+      setMessage("");
+    } catch (err) {
+      alert("Failed to send feedback. Please try again later.");
```
Comment (brian2w) [inline frontend/app/(footer)/suggestions/page.tsx:42]: log the error so we can have a more detailed stack trace

**Review by brian2w — CHANGES_REQUESTED** [review]
Looking good, just a few minor changes, and I think it should be good to go

File: frontend/observability/logger.ts (hunk 1 above)
Comment (ReggieStar11) [reply frontend/observability/logger.ts:8]: ya

File: frontend/app/(footer)/suggestions/page.tsx [hunk 3]
```diff
@@ -1,66 +1,117 @@
+"use client";
+
 import Logo from "@/components/navbar/Logo";
+import { sendEmail } from "@/services/src/emailJS/emailJS";
 import LightBulbIcon from "@/svgs/LightBulbIcon";
+import { useState } from "react";
+
+// Define the type for template parameters
+type TemplateParams = {
```
[the remark this answers is not in the document]
Comment (ReggieStar11) [reply frontend/app/(footer)/suggestions/page.tsx]: yea done

**Review by brian2w — APPROVED** [review]
lgtm

**Pull request merged**